sys/net/include/netdev.h

```diff
@@ -0,0 +1,262 @@
+                                         for the device as unsigned value in
+                                         host byte order */
+    NETDEV_OPT_ADDRESS_LONG,        /**< Longer hardware address for the device
+                                         (e.g. EUI-64) for the device as
+                                         unsigned value in host byte order */
+    NETDEV_OPT_TX_POWER,            /**< The output of the device in dB as
+                                         signed value in host byte order */
+    NETDEV_OPT_MAX_PACKET_SIZE,     /**< Maximum packet size the device supports
+                                         unsigned value in host byte order */
+    NETDEV_OPT_SRC_LEN,             /**< Default mode the source address is
+                                         set to as value of `size_t`. (e.g.
+                                         either PAN-centric 16-bit address or
+                                         EUI-64 in IEEE 802.15.4) */
+    NETDEV_OPT_EN_PRELOADING,       /**< Enable pre-loading of data, transfer
+                                         data to device using send_data(), send
+                                         by calling set_state(NETDEV_STATE_TX) */
+    NETDEV_OPT_EN_PROMISCUOUSMODE   /**< Enable promiscuous mode */
+    NETDEV_OPT_EN_AUTOACK,          /**< Automatically send link-layer ACKs */
+    NETDEV_OPT_RSSI,                /**< Read the RSSI value from the last transfer */
+    NETDEV_OPT_LQI,                 /**< Read the link quality indicator */
+
+    /**
+     * @brief   Last value for @ref netdev_opt_t defined here
+     *
+     * @details Specific devices or modules like @ref netapi that utilize these
+     *          values to may define higher values, but they must be greater
+     *          or equal to @ref NETDEV_OPT_LAST.
+     */
+    NETDEV_OPT_LAST,
+} netdev_opt_t;
+
+/**
+ * @brief   Definition of basic network device.
+ * @note    Feel free to expand if your device needs/supports more
+ */
+typedef enum {
+    NETDEV_ACTION_POWEROFF = 0,
+    NETDEV_ACTION_SLEEP,
+    NETDEV_ACTION_IDLE,
+    NETDEV_ACTION_RX,
+    NETDEV_ACTION_TX,               /**< only needed if pre-loading is enabled */
+} netdev_action_t;
+
+/**
+ * @brief   Definition of basic network device.
+ * @note    Feel free to expand if your device needs/supports more
+ */
+typedef enum {
+    NETDEV_STATE_OFF = 0,           /**< Device is powered off */
+    NETDEV_STATE_SLEEP,             /**< Device is sleeping */
+    NETDEV_STATE_IDLE,              /**< Device is idle */
+    NETDEV_STATE_RX,                /**< Device is in receive mode */
+    NETDEV_STATE_TX,                /**< Device is transmit mode */
+    NETDEV_STATE_PROMISCUOUS,       /**< Device is in receive mode and
+                                         accepts all packets without regard
+                                         for their destination */
```

Comment:
Please use multiline-comments according to coding conventions:

```C
    NETDEV_STATE_PROMISCUOUS,       /**< Device is in receive mode and
                                     *   accepts all packets without regard
                                     *   for their destination */
```

or don't document this member in-line.